src/bvals/comms/tag_map.cpp

```diff
@@ -45,13 +45,20 @@ void TagMap::AddMeshDataToMap(std::shared_ptr<MeshData<Real>> &md) {
 }
 
 void TagMap::ResolveMap() {
+  #ifdef MPI_PARALLEL
+  int flag;
+  void *max_tag; // largest supported MPI tag value
+  MPI_Comm_get_attr( MPI_COMM_WORLD, MPI_TAG_UB, &max_tag, &flag);
+  #endif
   for (auto it = map_.begin(); it != map_.end(); ++it) {
     auto &pair_map = it->second;
     int idx = 0;
     std::for_each(pair_map.begin(), pair_map.end(),
                   [&idx](auto &pair) { pair.second = idx++; });
-    if (idx > 32767)
-      PARTHENON_FAIL("Number of tags exceeds the maximum allowed by the MPI standard.");
+    #ifdef MPI_PARALLEL
+    if (idx > (*(int*)max_tag))
+      PARTHENON_FAIL("Number of tags exceeds the maximum allowed by this MPI version.");
+    #endif
```

Comment:
```suggestion
    #ifdef MPI_PARALLEL
    if (idx > (*reinterpret_cast<int*>(max_tag)) && it->first != Globals::my_rank)
      PARTHENON_FAIL("Number of tags exceeds the maximum allowed by this MPI version.");
    #endif
```

Reply:
Exceeding the maximum number of tags is fine if the sender and receiver are on the same rank (since MPI is not used in this case). The linter complains about the c-style cast. I think the `reinterpret_cast` is the right thing to do, but not totally sure.